out_writeapi.go

```diff
@@ -322,10 +325,9 @@ func FLBPluginFlushCtx(ctx, data unsafe.Pointer, length C.int, tag *C.char) int
 				log.Fatal("AppendRows: ", err)
 				return output.FLB_ERROR
 			}
+			config.mutex.Lock()
```

Comment:
This lock needs to cover both the call to managedStream.AppendRows and the subsequent call to appendResults.append(). That's because we need to make sure that the order in which the append results appear in the queue exactly matches the order in which they were sent. Note that as a consequence, you will also need to handle releasing the lock in case of an error.